engine-workspace/src/contract.rs

```diff
@@ -0,0 +1,393 @@
+    CallWithdraw, ViewAccountsCounter, ViewBalance, ViewBlockHash, ViewBridgeProver, ViewChainId,
+    ViewCode, ViewErc20FromNep141, ViewFtBalanceOf, ViewFtBalanceOfEth, ViewFtMetadata,
+    ViewFtTotalEthSupplyOnAurora, ViewFtTotalEthSupplyOnNear, ViewFtTotalSupply, ViewIsUsedProof,
+    ViewNep141FromErc20, ViewNonce, ViewOwner, ViewPausedFlags, ViewPausedPrecompiles,
+    ViewStorageAt, ViewStorageBalanceOf, ViewUpgradeIndex, ViewVersion, ViewView,
+};
+use crate::transaction::{CallTransaction, ViewTransaction};
+use aurora_engine_types::account_id::AccountId;
+use aurora_engine_types::parameters::connector::{FungibleTokenMetadata, Proof};
+use aurora_engine_types::parameters::engine::{
+    CallArgs, FunctionCallArgsV2, NewCallArgs, NewCallArgsV2,
+};
+use aurora_engine_types::parameters::xcc::FundXccArgs;
+use aurora_engine_types::types::{Address, RawU256, WeiU256};
+use aurora_engine_types::{H256, U256};
+use near_sdk::json_types::U128;
+use serde_json::json;
+
+#[derive(Debug, Clone)]
+pub struct EngineContract {
+    contract: RawContract,
+    pub node: Node,
+}
+
+impl EngineContract {
+    pub fn as_raw_contract(&self) -> &RawContract {
+        &self.contract
+    }
+
+    pub fn id(&self) -> AccountId {
+        self.contract.id()
+    }
+
+    pub fn root(&self) -> Account {
+        self.node.root()
+    }
+}
+
+impl From<(RawContract, Node)> for EngineContract {
+    fn from((contract, node): (RawContract, Node)) -> Self {
+        Self { contract, node }
+    }
+}
+
+/// Callable functions implementation.
+impl EngineContract {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new(
+        &self,
+        chain_id: RawU256,
+        owner_id: AccountId,
+        upgrade_delay_blocks: u64,
+    ) -> CallNew {
+        let args = NewCallArgs::V2(NewCallArgsV2 {
+            chain_id,
+            owner_id,
+            upgrade_delay_blocks,
+        });
+
+        CallNew::call(&self.contract).args_borsh(args)
+    }
+
+    pub fn new_eth_connector(
+        &self,
+        prover_account: AccountId,
+        custodian_address: String,
+        metadata: FungibleTokenMetadata,
+    ) -> CallNewEthConnector {
+        CallNewEthConnector::call(&self.contract).args_borsh((
+            prover_account,
+            custodian_address,
+            metadata,
+        ))
+    }
+
+    pub fn ft_transfer(
+        &self,
+        receiver_id: &AccountId,
+        amount: U128,
+        memo: Option<String>,
+    ) -> CallFtTransfer {
+        CallFtTransfer::call(&self.contract)
+            .args_json(json!({ "receiver_id": receiver_id, "amount": amount, "memo": memo }))
+    }
+
+    pub fn ft_transfer_call(
+        &self,
+        receiver_id: &AccountId,
+        amount: U128,
+        memo: Option<String>,
+        msg: String,
+    ) -> CallFtTransferCall {
+        CallFtTransferCall::call(&self.contract).args_json(json!({
+           "receiver_id": receiver_id,
+           "amount": amount.0.to_string(),
```

Comment:
We don't need to unwrap `U128` and convert it to `String`. `U128` serialized to `String` itself.

Reply:
Good catch.